wire-ios-data-model/Source/UseCases/ObserveMLSGroupVerificationStatusUseCase.swift

```diff
@@ -18,35 +18,35 @@
 
 import Foundation
 
-public protocol MLSConversationVerificationManagerInterface {
+public protocol ObserveMLSGroupVerificationStatusUseCaseProtocol {
 
-    func startObservingMLSConversationVerificationStatus()
+    func invoke()
 
 }
 
-public final class MLSConversationVerificationManager: MLSConversationVerificationManagerInterface {
+public final class ObserveMLSGroupVerificationStatusUseCase: ObserveMLSGroupVerificationStatusUseCaseProtocol {
 
     // MARK: - Properties
 
     var mlsService: MLSServiceInterface
-    var mlsConversationVerificationStatusProvider: MLSConversationVerificationStatusProviderInterface
+    var updateMLSGroupVerificationStatusUseCase: UpdateMLSGroupVerificationStatusUseCaseProtocol
 
     // MARK: - Life cycle
 
     public init(
         mlsService: MLSServiceInterface,
-        mlsConversationVerificationStatusProvider: MLSConversationVerificationStatusProviderInterface) {
+        updateMLSGroupVerificationStatusUseCase: UpdateMLSGroupVerificationStatusUseCaseProtocol) {
             self.mlsService = mlsService
-            self.mlsConversationVerificationStatusProvider = mlsConversationVerificationStatusProvider
+            self.updateMLSGroupVerificationStatusUseCase = updateMLSGroupVerificationStatusUseCase
         }
 
     // MARK: - Methods
 
-    public func startObservingMLSConversationVerificationStatus() {
+    public func invoke() {
```

Comment:
I see that you just restructured code.
I suggest to refactor this eventually.
First: The `invoke` method should be async, instead of having all its body inside a `Task` closure.
Second: The reference to the potentially never ending task is discarded right away, so we will never be able to cancel it. 

Follow-up questions: When and how many times to we invoke this?
What happens if the user logs out?

Reply:
I've added task cancellation in deinit, check it out please